pkg/operator/etcdmemberscontroller/etcdmemberscontroller.go

```diff
@@ -0,0 +1,208 @@
+		return err
+	}
+
+	healthyEtcdMembers, notStartedEtcdMembers, unhealthyMembers, unknownEtcdMembers := []string{}, []string{}, []string{}, []string{}
+
+	for _, m := range etcdMembers {
+		switch c.etcdClient.MemberStatus(m) {
+		case etcdcli.EtcdMemberStatusAvailable:
+			healthyEtcdMembers = append(healthyEtcdMembers, m.Name)
+		case etcdcli.EtcdMemberStatusNotStarted:
+			notStartedEtcdMembers = append(notStartedEtcdMembers, m.Name)
+		case etcdcli.EtcdMemberStatusUnhealthy:
+			unhealthyMembers = append(unhealthyMembers, m.Name)
+		case etcdcli.EtcdMemberStatusUnknown:
+			unknownEtcdMembers = append(unknownEtcdMembers, m.Name)
+		}
+	}
+
+	if len(unhealthyMembers) != 0 {
+		_, _, updateErr := v1helpers.UpdateStatus(c.operatorClient, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+			Type:    "EtcdMembersDegraded",
+			Status:  operatorv1.ConditionTrue,
+			Reason:  "UnhealthyMembers",
+			Message: fmt.Sprintf("%s members are unhealthy", strings.Join(unhealthyMembers, ",")),
+		}))
+		if updateErr != nil {
+			c.eventRecorder.Warning("EtcdMembersErrorUpdatingStatus", updateErr.Error())
+			return updateErr
+		}
+	} else {
+		_, _, updateErr := v1helpers.UpdateStatus(c.operatorClient, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+			Type:    "EtcdMembersDegraded",
+			Status:  operatorv1.ConditionFalse,
+			Reason:  "AsExpected",
+			Message: "No unhealthy members found",
+		}))
+		if updateErr != nil {
+			c.eventRecorder.Warning("EtcdMembersErrorUpdatingStatus", updateErr.Error())
+			return updateErr
+		}
+	}
+
+	if len(notStartedEtcdMembers) != 0 {
+		_, _, updateErr := v1helpers.UpdateStatus(c.operatorClient, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+			Type:    "EtcdMembersProgressing",
+			Status:  operatorv1.ConditionTrue,
+			Reason:  "MembersNotStarted",
+			Message: fmt.Sprintf("%s members have not started yet", strings.Join(notStartedEtcdMembers, ",")),
+		}))
+		if updateErr != nil {
+			c.eventRecorder.Warning("EtcdMembersErrorUpdatingStatus", updateErr.Error())
+			return updateErr
+		}
+	} else {
+		_, _, updateErr := v1helpers.UpdateStatus(c.operatorClient, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+			Type:    "EtcdMembersProgressing",
+			Status:  operatorv1.ConditionFalse,
+			Reason:  "AsExpected",
+			Message: "all members have started",
+		}))
+		if updateErr != nil {
+			c.eventRecorder.Warning("EtcdMembersErrorUpdatingStatus", updateErr.Error())
+			return updateErr
+		}
+	}
+
+	if len(healthyEtcdMembers) > len(etcdMembers)/2 {
+		_, _, updateErr := v1helpers.UpdateStatus(c.operatorClient, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+			Type:    "EtcdMembersAvailable",
+			Status:  operatorv1.ConditionTrue,
+			Reason:  "MembersNotStarted",
+			Message: fmt.Sprintf("%s members are available, %s have not started, %s are unhealthy", strings.Join(healthyEtcdMembers, ","), strings.Join(notStartedEtcdMembers, ","), strings.Join(unhealthyMembers, ",")),
```

Comment:
you forgot to display the unknown members

Reply:
done